server/middleware/filterTripId.js

```diff
@@ -0,0 +1,22 @@
+import jwt from 'jsonwebtoken';
+import resPonse from '../helpers/responses/response';
+import Trip from '../models/trips';
+
+const trip = new Trip();
+
+const allTrips = async (req, res, next) => {
+  const { tripId } = req.params;
+  const foundtrips = await trip.getTripById(tripId);
+  const tripData = foundtrips.rows;
+  const notadmin = jwt.decode(req.headers.authorization);
+  const activeTrips = tripData.filter(x => x.status === 'cancelled');
+  if (!req.headers.authorization || req.headers.authorization === ''
+  || notadmin.isadmin === false || notadmin === null){
```

Comment:
Could you write a reusable function which you just call here and other places that require authorization checks.